src/js/modules/board.js

```diff
@@ -0,0 +1,81 @@
+export function Board(size) {
+    if (!Number.isInteger(size) || size < 0) {
+        throw new TypeError(`invalid board size: ${size}`);
+    }
+    this._squares = new Array(size);
+    this._size = size;
+    this._hasPlayed = new Array();
+}
+
+Board.prototype.place = function(block, location) {
+    let coordinates = block.getContextualCoordinates(location);
+    coordinates.forEach((coord) => this._validatePlacement(coord, block.playerIndex));
+    coordinates.forEach((coord) => this._addSquare(coord, block.playerIndex));
+};
+
+Board.prototype._validatePlacement = function(coordinates, playerIndex) {
+    const [ x, y ] = coordinates;
+    if (coordinates.some((c) => this._isOffBoard(c))) {
+        throw TypeError(`block is placed off the board at [${x}, ${y}]`);
+    }
+    if (this._squares[x] !== undefined && this._squares[x][y] !== undefined) {
+        throw TypeError(`block already placed at [${x}, ${y}]`);
+    }
+    if (this._sameAdjacentPlayer(x, y, playerIndex)) {
+        throw TypeError(`block placed adjacent to same player [${x}, ${y}]`);
+    }
+    if (!this._hasPlayed[playerIndex] && !this._isStartingPlacment(coordinates, playerIndex)) {
+        throw TypeError(`first play by player at invalid location [${x}, ${y}]`);
+    }
+};
```

Comment:
we may want a version  of this that returns a bool so we don't need to try/catch during a game